From 65667f870a79f56cff754ff4a2395bf663648e93 Mon Sep 17 00:00:00 2001
From: "A. Cody Schuffelen" <schuffelen@google.com>
Date: Wed, 29 Apr 2020 14:52:49 -0700
Subject: [PATCH 08/15] Add host_supported to some Keymaster libraries.

This will allow building PureSoftKeymaster on the host.

"-fno-rtti" is added for parity with the `stl: "none"` setting, as rtti
doesn't work without the c++ standard library, producing some errors
like

ld.lld: error: undefined symbol: vtable for __cxxabiv1::__class_type_info
ld.lld: error: undefined symbol: vtable for __cxxabiv1::__si_class_type_info
ld.lld: error: undefined symbol: vtable for __cxxabiv1::__vmi_class_type_info

Test: Compiles
Bug: 155328339
Change-Id: I1975f4ece13aac3cc5882262ca1a30dc1f615150
---
 Android.bp | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index 3ce107c..fb69928 100644
--- a/Android.bp
+++ b/Android.bp
@@ -54,6 +54,14 @@ cc_library_shared {
     ],
     stl: "none",
     export_include_dirs: ["include"],
+    host_supported: true,
+    target: {
+        host: {
+            clang_cflags: [
+                "-fno-rtti", // TODO(b/156427382): Remove workaround when possible.
+            ],
+        },
+    },
 }
 
 // libkeymaster_portable contains almost everything needed for a keymaster
@@ -124,7 +132,15 @@ cc_library {
     // weakly defines the subset of stl symbols required for this library to work
     // and which are also available in the trusty context.
     stl: "none",
+    host_supported: true,
     export_include_dirs: ["include"],
+    target: {
+        host: {
+            clang_cflags: [
+                "-fno-rtti", // TODO(b/156427382): Remove workaround when possible.
+            ],
+        },
+    },
 }
 
 // libsoftkeymaster provides a software-based keymaster HAL implementation.
@@ -172,6 +188,7 @@ cc_library {
         "libkeymaster_portable",
     ],
 
+    host_supported: true,
     export_include_dirs: ["include"],
 }
 
@@ -193,7 +210,6 @@ cc_library {
         "libcutils",
         "libbase",
     ],
-
     export_include_dirs: ["include"],
 }
 
-- 
2.17.1

